### PR TITLE
feat : Added product listing page and filter functionality

### PR DESCRIPTION
**Pull request by aiman-ansari**
Files to check
- Product.js
- Component/Product
- Context/DataContext

### CODE REVIEW

File: src/Components/Product/Filter.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            <hr />
+            <div class="pd-1">
+                <p class="text-dark">Price</p>
+                    <input 
+                        type="range"
+                        min="1000"
+                        max="15000"
+                        step="20"
+                        value={price}
+                        label={price}
```
Comment (junaidshaikh-js) [inline src/Components/Product/Filter.js]: `input` elements do have a `label` attribute?

File: src/Components/Product/Filter.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                            id="high-to-low" 
+                            onChange = {() =>
+                                dispatch({
+                                    type:'SORT_BY_PRICE',
+                                    payload: 'HighToLow'
+                                })}
+                            checked={sort === 'HighToLow' ? true :  false}
+                        >
+                        </input>
+                        <span>Price - High to Low</span>
```
Comment (junaidshaikh-js) [inline src/Components/Product/Filter.js:54]: wrap the text inside `label` instead of `span`

File: src/pages/Product/Product.js [hunk 3]
```diff
@@ -1,7 +1,25 @@
+import { Filter } from "../../Components/Product/Filter";
+import { SingleProducts , FilterProduct } from "../../Components/Product";
+import { useData } from "../../Context/DataContext/DataContext";
+import '../../Components/Product/Product.css'
+
 export default function Product(){
+    const {product, state } = useData()
+    let allSortedProducts = FilterProduct(product, state)
+    console.log(allSortedProducts)
```
Comment (junaidshaikh-js) [inline src/pages/Product/Product.js]: remove console logs

**Review by junaidshaikh-js — COMMENTED** [review]
Great Work! 

Good folder structure and clean code! Can be merged.

File: src/Components/Product/Filter.js (hunk 1 above)
Comment (aiman-ansari) [reply src/Components/Product/Filter.js]: I forgot to remove it thanks for pointing it out.

**Review by sanu29 — APPROVED** [review]
All good to proceed with commit according to me. Great work could not find any issues.
Spacing is great as well your naming conventions are meaningful and easy.
File structure is also good. Code is readable and clean.
Great work.

**Pull request merged**